### PR TITLE
feat: create "biome" hook, preserve "rome" hook as alias

### LINKED ISSUE #428: `biome` is being used instead of `rome`
While navigating the repository to implement https://github.com/cachix/git-hooks.nix/issues/425, I noticed that the `rome` hook is already configured to use `biome` instead of the actual `rome` tool.

This changed was performed in https://github.com/cachix/git-hooks.nix/pull/391, and `biome` is still referenced and used as `rome` throughout the documentation and code.

I propose to rename the `rome` hook to `biome` and create an alias to `rome` for backwards compatibility.

### PR DESCRIPTION
**Pull request by oxcabe**
This PR does the following:
* Renames the "rome" hook to "biome", correcting any references to the rome tool.
* Sets the "rome" hook as an alias to the "biome" hook as a backwards compatibility guard.
* Adds relevant documentation to the README.

Fixes #428.

### CODE REVIEW

**Review by sandydoo — CHANGES_REQUESTED** [review]
@oxcabe, could you please remove all the code formatting changes? The diff should be readable and contain just the changes you're proposing. Thanks!

**Comment by oxcabe** [comment]
Sure thing, I'm not even sure when the files got formatted :sweat_smile:

**Review by sandydoo — CHANGES_REQUESTED** [review]
I think the better approach here would be to use `mkRenamedOptionModule` to rename `rome -> biome`. You get the alias and people will be prompted to switch.

If you feel like an alias (with no warning) is more appropriate for now, then you can use `mkAliasOptionModule` instead. 

There's no need to have both `rome` and `biome` in `config.hooks`.

**Comment by oxcabe** [comment]
I know it's failing as of my last commit. Currently trying to figure out by myself what am I doing wrong and why 👍🏼

**Comment by oxcabe** [comment]
> I think the better approach here would be to use mkRenamedOptionModule to rename rome -> biome. You get the alias and people will be prompted to switch.

Implemented this approach. For some reason, the deprecation warning is not appearing when enabling the hook:

```
rome.enable = true;
```

> There's no need to have both rome and biome in config.hooks.

I can't seem to make it work without having `config.hooks` entries for both tools.

File: modules/hooks.nix [hunk 1]
```diff
@@ -1638,7 +1640,7 @@ in
     };
```
Comment (sandydoo) [inline modules/hooks.nix:1641]: Ah, okay, we're hitting the limitations of the freetype+submodule system. We should manually add a warning to help people switch over to biome. Something like this:

```suggestion

  config.warnings =
    lib.optional cfg.hooks.rome.enable ''
      The hook `hooks.rome` has been renamed to `hooks.biome`.
    '';

```

**Review by sandydoo — CHANGES_REQUESTED** [review]

**Comment by oxcabe** [comment]
From the failing build:

```
pre-commit-run> - hook id: nixpkgs-fmt
pre-commit-run> - files were modified by this hook
```

File: modules/hooks.nix (hunk 1 above)
Comment (oxcabe) [reply modules/hooks.nix:1641]: I like this approach :+1:

**Comment by oxcabe** [comment]
Ready!

**Review by sandydoo — APPROVED** [review]

**Pull request merged**